### PR TITLE
Refactoring QueryPipeline

### PR DESCRIPTION
**Pull request by amosbird**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove duplicated code in QueryPipeline.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/QueryPipeline/QueryPipelineBuilder.h [hunk 1]
```diff
@@ -159,6 +159,7 @@ class QueryPipelineBuilder
     /// Convert query pipeline to pipe.
     static Pipe getPipe(QueryPipelineBuilder pipeline) { return std::move(pipeline.pipe); }
     static QueryPipeline getPipeline(QueryPipelineBuilder builder);
+    static void initRowsBeforeLimit(IOutputFormat * output_format);
```
Comment (antonio2368) [inline src/QueryPipeline/QueryPipelineBuilder.h:162]: The non static declaration can be deleted now right?

Comment (amosbird) [reply src/QueryPipeline/QueryPipelineBuilder.h:162]: What do you mean by "non static declaration" ?

**Review by KochetovNicolai — APPROVED** [review]

File: src/QueryPipeline/QueryPipelineBuilder.h (hunk 1 above)
Comment (amosbird) [reply src/QueryPipeline/QueryPipelineBuilder.h:162]: Ah, I see.

**Comment by antonio2368** [comment]
@Mergifyio update

**Review by antonio2368 — APPROVED** [review]

**Comment by antonio2368** [comment]
@KochetovNicolai anything more to delete or we good for merge? 😄

**Pull request merged**